src/main/kotlin/com/sparetimedevs/consonance/routes/UsersRoute.kt

```diff
@@ -0,0 +1,52 @@
+package com.sparetimedevs.consonance.routes
+
+import com.sparetimedevs.consonance.model.User
+import com.sparetimedevs.consonance.repository.UserRepository
+import com.sparetimedevs.consonance.routes.misc.ID
+import com.sparetimedevs.consonance.routes.misc.ID_PATH_PARAM
+import com.sparetimedevs.consonance.routes.misc.toObjectId
+import io.ktor.application.call
+import io.ktor.http.HttpStatusCode
+import io.ktor.request.receive
+import io.ktor.response.respond
+import io.ktor.routing.Route
+import io.ktor.routing.delete
+import io.ktor.routing.get
+import io.ktor.routing.post
+import io.ktor.routing.put
+import mu.KotlinLogging
+import org.koin.ktor.ext.inject
+
+private val logger = KotlinLogging.logger {}
+
+const val USERS_PATH = "/users"
+
+fun Route.userRoutes() {
+
+    val userRepository: UserRepository by inject()
```

Comment:
You have a point that indeed changing the DI framework would touch multiple classes, however i only used the by inject() style in the Route extension functions due to koin providing specific support for the ktor framework and this being their recommended way of using it. All normal (non route) classes are using the regular, non ktor-koin pattern of constructor injection. Refer to [](https://insert-koin.io/docs/1.0/getting-started/ktor/) to see how they recommend using it. In my opinion, this is pretty clean and readable, though i grant you the point that it is coupling us to koin and therefore koin test. I think it's unlikely we would switch over to kodeine( or another DI framework). If you look at kodeine, it also has the same pattern of the code knowing about the framework. [https://github.com/Kodein-Framework/Kodein-DI](url)

Reply:
Maybe I'm not looking hard enough, but I do not see the Koin docs suggesting to inject in multiple places. Only in the `fun Application.main()`. Now I do realise there can be a point made for also injecting into `Route`, because you do not have access to the constructor. Let's discuss in person. :)

Reply:
I've approved this PR. Personally, I do not like the coupling between Koin DI framework and the Route class. But I also see the simplicity of this solution. We can always decide later to change it one way or another.

Reply:
Sounds good champ. Thanks.